app/master/cluster_master.py

```diff
@@ -279,7 +282,7 @@ def handle_result_reported_from_slave(self, slave_url, build_id, subjob_id, payl
         build = self._all_builds_by_id[int(build_id)]
         slave = self._all_slaves_by_url[slave_url]
         # If the build has been canceled, don't work on the next subjob.
-        if not build.is_finished:  # WIP(joey): This check should be internal to the Build object.
+        if not build.is_finished or build.has_error:  # WIP(joey): This check should be internal to the Build object.
```

Comment:
@ethomas2 The bug TJ was referring to I think is a little more involved:
1. Slave A is allocated to build. Setup succeeds. First subjobs are sent to slave A.
2. Slave B is allocated to build. Setup fails. Build is marked failed (`build.has_error == True`).
3. Subjob results are reported from slave A. `handle_result_reported_from_slave()` is a no-op since `build.is_finished` is false. Slave A is stuck since `execute_next_subjob_or_free_executor` is never called.